Improve MotorEx Velocity and Motor Configuration Type

JIceberg opened the pull request:
# Improve MotorEx Velocity and Motor Configuration Type

Please note that we accept pull requests from anyone, but that does not mean it will be merged.

## What kind of change does this PR introduce?
* Feature

Adds a buffer to the motors for velocity PID mode.
In `MotorEx`, setting the velocity now calls `set()` and `set()` calls `setPower()` for `VelocityControl`.
Streamlines the constructors and sets the RPM and CPR of the motor in the configuration type for the third constructor.

## Did this PR introduce a breaking change?
* No

__Please make sure your PR satisfies the requirements of [the contributing page](CONTRIBUTING.md)__

NoahAndrews commented:
I'm not very familiar with this library and its internals (I'm just watching it on GitHub so I can have an idea of what you're doing), so maybe this PR doesn't do what I think it might. But if you're entirely eliminating the buffer between the theoretical max velocity, and the velocity that's requested when the user specifies a power of 1.0, you probably should not do that.

A robot under load will not be able to reach that maximum velocity, so the motor will just end up running as fast as it can, negating the benefits of closed-loop control. You even want to leave some buffer with the max velocity under load in most cases, so that the motor can better maintain its velocity when there is extra friction or load. 

--Noah from REV

JIceberg commented:
> I'm not very familiar with this library and its internals (I'm just watching it on GitHub so I can have an idea of what you're doing), so maybe this PR doesn't do what I think it might. But if you're entirely eliminating the buffer between the theoretical max velocity, and the velocity that's requested when the user specifies a power of 1.0, you probably should not do that.
> 
> A robot under load will not be able to reach that maximum velocity, so the motor will just end up running as fast as it can, negating the benefits of closed-loop control. You even want to leave some buffer with the max velocity under load in most cases, so that the motor can better maintain its velocity when there is extra friction or load.
> 
> --Noah (from REV)

Good points. Yeah the PR was eliminating the buffer. Previously we were never even using RUN_USING_ENCODER and already performed calculations through an external PIDF controller (not on-board), and I added this so that if the user accidentally entered RUN_USING_ENCODER mode. What if, instead, we implemented a user-manageable buffer similar to modifying the configuration type? i.e.
```java
public void setBufferFraction(float buffer) {
  bufferFraction = buffer;
}

public void set(double output) {
  if (mode == RunMode.VelocityControl) {
    double speed = output * ACHIEVABLE_MAX_TICKS_PER_SECOND * bufferFraction;
    ...
  }
}
```
and then use that as the target speed?

NoahAndrews commented:
Hmm, interesting. Seems reasonable enough.

puravdatta-sudo reviewed (approved):
Looks good, Is there a point in doing the 0.9 as a buffer rather than the default 0.9

JIceberg commented:
> Looks good, Is there a point in doing the 0.9 as a buffer rather than the default 0.9

Well, the default in the SDK is 0.85, not 0.9. I chose 0.9 arbitrarily. Seems to run just fine with it, haven't seen any issues except under a bit larger than average load from a few of my tests (in which it starts to try to run at max speed and fails to maintain the benefit of the controller).
